faros_event.sh

```diff
@@ -0,0 +1,470 @@
+#
+# Required args:
+# Arg  | Description
+# -----------------------------------------------------------------------------
+# arg1 | Emit type: "deployment", "build", "full" (must be first arg)
+# 
+# Required fields:
+# Flag                           | Env Var
+# -----------------------------------------------------------------------------
+# -k / --api_key <api_key>       | FAROS_API_KEY
+# -a / --application <app_name>  | APPLICATION_NAME
+# -c / --commit <commit_sha>     | COMMIT_SHA
+# -p / --pipeline <pipeline>     | PIPELINE_UID
+# --ci_org <ci_org>              | CI_ORG_UID
+#
+# (Required deployment fields)
+# -e / --environment <env>       | DEPLOYMENT_ENV
+# -ds / --deploy_status <status> | DEPLOYMENT_STATUS (e.g. Success, Failed, etc)
+#
+# (Required build fields)
+# -bs / --build_status <status>  | BUILD_STATUS (e.g. Success, Failed, etc)
+# -r / --repo <repo>             | REPOSITORY
+# --vcs_org <vcs_org>            | VCS_ORG_UID
+# --vcs_source <vcs_source>      | VCS_SOURCE
+#
+# Optional fields:
+# Flag                  | Env Var                  | Default
+# -----------------------------------------------------------------------------
+# -g / --graph <graph>  | FAROS_GRAPH              | "default"
+# -url <url>            | FAROS_API_URL            | https://prod.api.faros.ai
+#                       | ORIGIN                   | "Faros_Script_Event"
+#                       | SOURCE                   | "Faros_Script"
+#                       | START_TIME               | Now (e.g. 1549932180000)
+#                       | END_TIME                 | Now (e.g. 1549932180000)
+#
+# (Optional deployment fields)
+#                       | DEPLOYMENT_UID           | Random UUID
+#                       | APPLICATION_PLATFORM     | "NA"
+#                       | DEPLOYMENT_ENV_DETAIL    | ""
+#                       | DEPLOYMENT_STATUS_DETAIL | ""
+#                       | DEPLOYMENT_START_TIME    | START_TIME
+#                       | DEPLOYMENT_END_TIME      | END_TIME
+#
+# (Optional build fields)
+#                       | BUILD_UID                | Commit Sha
+#                       | BUILD_STATUS_DETAIL      | ""
+#                       | BUILD_START_TIME         | START_TIME
+#                       | BUILD_END_TIME           | END_TIME
+#
+# Optional Script Flags:
+# Flag          | Description
+# -----------------------------------------------------------------------------
+# --dry_run     | If present, the event will be printed instead of emitted.
+# --print_event | If present, the event will be printed.
+# --debug       | If present, helpful information will be printed.
+main() {
+    EMIT_TYPE=$1
+    shift
+
+    parseArgs "$@"  
+    validateInput
+
+    if ((debug))
+    then
+        echo "Faros url: $faros_api_url"
+        echo "Faros graph: $faros_graph"
+        echo "Dry run: $dry_run"
+        echo "Print event: $print_event"
+        echo "Debug: $debug"
+    fi
+
+    if [ $EMIT_TYPE = "deployment" ]
```

Comment:
Please change emit to event, e.g. `EMIT_TYPE` -> `EVENT_TYPE`. Here and elsewhere in the code and docs.